### PR TITLE
test: initial integration test for knative related functionality

### PR DESCRIPTION
**Pull request by matejvasek**
# Changes

- :test_tube: Initial integration tests for `knative` related functionality.

Related: https://github.com/knative/func/issues/1278

### CODE REVIEW

**Comment by matejvasek** [comment]
@dolfolife PTAL

File: knative/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	fn "knative.dev/func"
+	"knative.dev/func/k8s"
+	"knative.dev/func/knative"
+)
+
+func TestIntegration(t *testing.T) {
```
Comment (lance) [inline knative/integration_test.go:23]: This looks great! Looks like this change is great for our coverage. But would it not be better in the long run to have individual tests for deploy, describe and list? I know that means deploying three separate times, so maybe it's overkill.

Comment (matejvasek) [reply knative/integration_test.go:23]: I want to test that all the operation works together.

**Pull request merged**